joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -221,6 +246,39 @@ JointTrajectoryController::on_configure(const rclcpp_lifecycle::State & previous
   // TODO(karsten1987): no lifecyle for subscriber yet
   // joint_command_subscriber_->on_activate();
 
+  // State publisher
+  {
+    int state_publish_rate = lifecycle_node_->get_parameter("state_publish_rate").get_value<int>();
+    RCLCPP_INFO_STREAM(
+      logger, "Controller state will be published at " <<
+        state_publish_rate << "Hz.");
+    if (state_publish_rate > 0) {
+      state_publisher_period_ =
+        rclcpp::Duration::from_seconds(1.0 / static_cast<double>(state_publish_rate));
+    } else {
+      state_publisher_period_ = rclcpp::Duration(0.0);
```

Comment:
In `JointTrajectoryController::update()` we check `if (state_publisher_period_.seconds() > 0.0)`.
This line here means we won't perform any update when a negative rate is passed. Is that the intended behavior?

Reply:
Yes. That being said, we could improve it to throw an error, though inputting a negative rate is so bewildering that we can say whoever writes it probably wants to save keystrokes or something. I'd allow that flexibility.

Reply:
I'd advocate for printing an error message and return an error here if negative. It's clearly negatively configured and the lifecycle should not continue in this case.